src/librustc_resolve/diagnostics.rs

```diff
@@ -838,6 +838,31 @@ trait Foo {
 
 fn foo<T>(x: T) {} // ok!
 ```
+
+Another case that causes this error is when a type is imported into a parent
+module. To fix this, you can follow the suggestion and use File directly or
+`use super::File;` which will import the types from the parent namespace. An
+example that causes this error is below:
+
+```
+use std::fs::File;
+
+mod foo {
+    fn some_function(f: File) {}
+}
+```
+
+```
+use std::fs::File;
+
+mod foo {
+    // either
+    use super::File;
```

Comment:
The problem here was that rustdoc auto-injects `fn main` surrounding this, so it was correct -- there was no `File` in root. I inserted an empty commented fn main to workaround this, so we should be fine now.